From f463f2dc8b8e2d6c705fd558873874d0b4116f3f Mon Sep 17 00:00:00 2001
From: Mike Looijmans <mike.looijmans@topic.nl>
Date: Wed, 19 Feb 2020 11:43:43 +0100
Subject: [PATCH 30/30] devicetree: zynqmp.dtsi: Add bootmode selection support

Add bootmode override support for ZynqMP devices. Allows one to select
a boot device by running "reboot qspi32" for example. Activate config
item CONFIG_SYSCON_REBOOT_MODE to make this work.

Signed-off-by: Mike Looijmans <mike.looijmans@topic.nl>
---
 arch/arm64/boot/dts/xilinx/zynqmp.dtsi | 24 ++++++++++++++++++++++++
 1 file changed, 24 insertions(+)

diff --git a/arch/arm64/boot/dts/xilinx/zynqmp.dtsi b/arch/arm64/boot/dts/xilinx/zynqmp.dtsi
index db61b4dfe670..2dd5ea9c25e6 100644
--- a/arch/arm64/boot/dts/xilinx/zynqmp.dtsi
+++ b/arch/arm64/boot/dts/xilinx/zynqmp.dtsi
@@ -347,6 +347,30 @@
 			};
 		};
 
+		/* Clock and Reset control registers for LPD */
+		lpd_apb: apb@ff5e0000 {
+			compatible = "syscon", "simple-mfd";
+			reg = <0x0 0xff5e0000 0x0 0x400>;
+			reboot-mode {
+				compatible = "syscon-reboot-mode";
+				offset = <0x200>;
+				mask = <0xf100>;
+				/* Bit(8) is the "force user" bit */
+				mode-normal = <0x0000>;
+				mode-psjtag = <0x0100>;
+				mode-qspi24 = <0x1100>;
+				mode-qspi32 = <0x2100>;
+				mode-sd0    = <0x3100>;
+				mode-nand   = <0x4100>;
+				mode-sd1    = <0x6100>;
+				mode-emmc   = <0x6100>;
+				mode-usb0   = <0x7100>;
+				mode-pjtag0 = <0x8100>;
+				mode-pjtag1 = <0x9100>;
+				mode-sd1ls  = <0xe100>;
+			};
+		};
+
 		/* GDMA */
 		fpd_dma_chan1: dma@fd500000 {
 			status = "disabled";
-- 
2.17.1

